internal/factsengine/gatherers/sapcontrol.go

```diff
@@ -89,24 +91,58 @@ type SapControlInstance struct {
 type SapControlGatherer struct {
 	webService sapcontrolapi.WebServiceConnector
 	fs         afero.Fs
+	cache      *factscache.FactsCache
 }
 
 func NewDefaultSapControlGatherer() *SapControlGatherer {
 	webService := sapcontrolapi.WebServiceUnix{}
 	fs := afero.NewOsFs()
-	return NewSapControlGatherer(webService, fs)
+	return NewSapControlGatherer(webService, fs, nil)
 }
 
-func NewSapControlGatherer(webService sapcontrolapi.WebServiceConnector, fs afero.Fs) *SapControlGatherer {
+func NewSapControlGatherer(
+	webService sapcontrolapi.WebServiceConnector,
+	fs afero.Fs,
+	cache *factscache.FactsCache) *SapControlGatherer {
+
 	return &SapControlGatherer{
 		webService: webService,
 		fs:         fs,
+		cache:      cache,
+	}
+}
+
+func (s *SapControlGatherer) SetCache(cache *factscache.FactsCache) {
+	s.cache = cache
+}
+
+func memoizeSapcontrol(args ...interface{}) (interface{}, error) {
```

Comment:
I cannot add a test for this...
In reality, this `!ok` shouldn't never happen as this function is only called internally with known arguments.
But I just added the cases for some unexpected error, that I cannot forsee now